syft/source/unindexed_directory_resolver.go

```diff
@@ -0,0 +1,565 @@
+package source
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/afero"
+	"golang.org/x/exp/slices"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+type WritableFileResolver interface {
+	FileResolver
+
+	Write(path string, reader io.Reader) error
```

Comment:
resolvers tend to deal with `Locations` instead of string paths in the content fetching, which is the closest use case to this. Is this something that was considered during the design process? (it may not make sense given the use case)